hedgehog/src/Hedgehog/Internal/Property.hs

```diff
@@ -449,6 +469,13 @@ liftExceptT :: (Monad m, Show x, HasCallStack) => ExceptT x m a -> Test m a
 liftExceptT m =
   withFrozenCallStack liftEither =<< lift (runExceptT m)
 
+-- | Fails the test if the 'ExceptT' is 'Left', otherwise returns the value in
+--   the 'Right'.
+--
+withExceptT :: (Monad m, Show x, HasCallStack) => Test (ExceptT x m) a -> Test m a
+withExceptT m =
+  withFrozenCallStack liftEither =<< runExceptT (distribute m)
```

Comment:
I'm aware that this clashes with `Control.Monad.Trans.Except.withExceptT` but I'm not sure what else to call it. Whatever it's called it should be the same prefix as `withResourceT` I think.